src/status_im/keycard/simulated_keycard.cljs

```diff
@@ -402,28 +407,29 @@
        wallet-root-address
        hashed-password
        (fn [value]
-         (let [{:keys [id error]} (types/json->clj value)]
-           (if error
-             (re-frame/dispatch [::new-account-error :password-error error])
-             (status/multiaccount-derive-addresses
-              id
-              [path]
-              (fn [derived]
-                (let [derived-address (get-in (types/json->clj derived) [(keyword path) :address])]
-                  (if (some (fn [a] (= derived-address (get a :address))) accounts)
-                    (re-frame/dispatch [::new-account-error :account-error
-                                        (i18n/label :t/account-exists-title)])
-                    (status/multiaccount-store-derived
-                     id
-                     key-uid
-                     [path]
-                     hashed-password
-                     (fn [result]
-                       (let [{:keys [error] :as result} (types/json->clj result)
-                             {:keys [publicKey]}        (get result (keyword path))]
-                         (if error
-                           (on-failure error)
-                           (on-success publicKey))))))))))))))))
+         (let [{:keys [id]} (types/json->clj value)]
+           (status/multiaccount-derive-addresses
+            id
+            [path]
+            (fn [derived]
+              (let [derived-address (get-in (types/json->clj derived) [(keyword path) :address])]
+                (if (some (fn [a] (= derived-address (get a :address))) accounts)
+                  (re-frame/dispatch [::new-account-error :account-error
+                                      (i18n/label :t/account-exists-title)])
+                  (status/multiaccount-store-derived
+                   id
+                   key-uid
+                   [path]
+                   hashed-password
+                   (fn [result]
+                     (let [{:keys [publicKey]} (get result (keyword path))]
+                       (on-success publicKey)))
+                   (fn [error-message]
+                     (on-failure error-message))))))
+            (fn [error-message]
+              (log/debug "error while status/multiaccount-derive-addresses" error-message)))))
```

Comment:
@siddarthkay In this PR I see a lot of `log/debug` in *error callback* functions. They seem to swallow the error, because they don't rethrow it. Are the errors really propagating up the call tree? If not, then I think this is a more serious problem.

If they propagate correctly, then it shouldn't be necessary to add intermediary calls to `log/debug`, only a top level error handler would be enough. In Java, this would be equivalent of doing a try/catch in intermediary methods that don't actually do anything with the error (except logging). This is a known anti-pattern.

It's probably out of scope for your PR, but these nested callbacks for success/error could be dramatically improved in the future if we used the de facto standard in Javascript (promises), or some other means in Clojure land. This would give us a sane way to handle errors and create composable error handlers.

Reply:
> dramatically improved in the future if we used the de facto standard in Javascript (promises)

I did look into the native modules side on React Native and it does support Promises and would make the code a lot simpler but would require a lot of re-write ( same as this PR 🫣 ).

I would for now log a new issue to address this concern later.

Reply:
> Are the errors really propagating up the call tree? If not, then I think this is a more serious problem.

yes they do and that was the inspiration to introduce error-callbacks to avoid handling of both success and error cases in one callback.

Reply:
> I would for now log a new issue to address this concern later.

Thanks, better in a separate issue as you said.